### ISSUE TITLE
Entities are not converted to pascal case when all tables are renamed using json config 

### ISSUE THREAD

**Issue by kedarrkulkarni** [ErikEJ/EFCorePowerTools#1440]
First of all, thanks for the brilliant tool. it works quite well.

All of my tables have name in capital letter, separated by "_" for example 

- PREFIX_PURPOSE_SUMMARY
- PREFIX_ANOTHER_PURPOSE_SUMMARY

If I use the PowerTool or EF core scaffold without .renaming.json then it converts all DBSet names in pascal case correctly as below

- PrefixPurposeSummary
- PrefixAnotherPurposeSummary

Since all of my tables have same prefix (don't ask me why), I want the generated entities to not have this prefix, so I used .renaming.json with following config

```
[
  {
    "SchemaName": "SchemaName",
    "UseSchemaName": false,
    "TablePatternReplaceWith": "",
    "TableRegexPattern": "((^PREFIX)?([ _,]+))"
  }
]
```
This works, and I get the entities generated without the prefix (great!), however the resultant entity names are not converted into PascalCase, so my generated entities now look as below

- PURPOSESUMMARY
- ANOTHERPURPOSESUMMARY

In summary, pascal casing is not applied if table pattern replacement is used with regex.
It looks like, this is a minor change in the code, I am happy to submit pull request if you want, however please let me know if I have got the configuration wrong?

### Provide technical details

- EF Core version in use: EF Core 6

- Is Handlebars used: no

- Is .dacpac used: no

- EF Core Power Tools version: 2.5.1007.0

- Database engine: Oracle

- Visual Studio version: Visual Studio 2019

**Comment by ErikEJ** [comment]
Feel free to submit a PR

**Comment by ErikEJ** [comment]
I think this would be a breaking change, so it must be opt-in

**Comment by ErikEJ** [comment]
Fixed in latest daily build, please try it out and let me know if that solves your issue

**Comment by kedarrkulkarni** [comment]
The fix works as expected, I have the models generated in pascal case, along with renaming regex.
Thanks for the quick fix and providing a tool that just works

**Comment by ErikEJ** [comment]
The code does actually not pascal case it just sanitizes.

Feel free to post a review on VS Marketplace

**Comment by kedarrkulkarni** [comment]
Thanks @ErikEJ 
The reason I originally raised the pascal case issue is because, if I don't use the efpt.renaming.json then the table PREFIX_PURPOSE_SUMMARY is generated as PrefixPurposeSummary (correct pascal case)

I was using efpt.reaming.json with TableRegexPattern to remove prefix from the entity name, which does actually remove the prefix but that's when it stopped the pascal conversion of the entity name.

As you said, if the entity names with vanilla PowerTool (without etft.remaming.json) was also not converting the entity names into pascal case then it would have been fine... however, this looks like the pascal case conversion is not being done only when reaming.json is used with TableRegexPattern. 

I was thinking of option to opt in pascal case conversion like below

[
  {
    "SchemaName": "SchemaName",
    "UseSchemaName": false,
    "TablePatternReplaceWith": "",
    **"TablePatternUsePascalCase": true,**
    "TableRegexPattern": "((^PREFIX)?([ _,]+))"
  }
]

**Comment by ErikEJ** [comment]
It is not **pascal cased**, just sanitized, but I get what you mean.

>> After the recent fix, it's being generated as Purposesummary

I think that may be because your regex removes ALL _ (underscores) where as it should just remove "PREFIX_" ?

**Comment by ErikEJ** [comment]
I found a bug in the current fix.. (And added a test with a correct regex for your scenario)

**Comment by oselly** [comment]
> I think this would be a breaking change, so it must be opt-in

Did this get changed to be Opt-in/Opt-out? 

We've got a similar setup where all our SQL tables are prefixed with tbl i.e. [schema].[tblSIMCard] but we use the renaming.json to remove the tbl prefix. Noticed today when running EFPowerTools Edit from a dacpac, instead of generating the class as SIMCard which it did previously, it's generating it as Simcard which is then breaking all the references to those classes. We've got quite a few other instances where we've got capitalised acronyms in the table name and it's doing the same thing. Preferably we'd like to keep the Acronyms capitalised/un sanitized like they were previously

Thanks

**Comment by oselly** [comment]
> You should be able to override by adding a renamer for the affected tables where name and newname are the same.

Sorted thanks for the speedy reply. Previously posted json would change to be the following for anyone else who stumbles across this having the same issue.

{
"ColumnPatternReplaceWith": "",
"ColumnRegexPattern": "^(tbl)",
"SchemaName": "SchemaName",
"TablePatternReplaceWith": "",
"TableRegexPattern": "^(tbl)",
"Tables": [
          {
            "Columns": [ ],
            "Name": "tblSIMCard",
            "NewName": "SIMCard"
         }
],
"UseSchemaName": false
}
